jujugui/static/gui/src/app/app.js

```diff
@@ -2116,15 +2114,30 @@ YUI.add('juju-gui', function(Y) {
       situations where auth is set outside the GUI (i.e., embedded).
 
       @method _getAuth
-      @param {String} service The service to retrieve auth info from.
      */
-    _getAuth: function(service) {
+    _getAuth: function() {
       var externalAuth = this.get('auth');
       if (externalAuth) {
         return externalAuth;
       }
       var users = this.get('users');
-      return users && users[service];
+      var user;
+      if (users) {
+        var controllerUser;
+        if (this.env) {
```

Comment:
I think this section of code may also be triggered before an env connection is established. I'll verify if that's the case. If it's just tests, then I'll see how painful it will be to fix, but these are the old tests so... mocking. Ugh.